src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -27,6 +38,91 @@ export interface RefetchQuoteCallbackParmams {
 
+// const FEE_EXCEEDS_FROM_ERROR = new QuoteError({
+//   errorType: QuoteErrorCodes.FeeExceedsFrom,
+//   description: QuoteError.quoteErrorDetails.FeeExceedsFrom
+// })
+
+class PriceQuoteError extends Error {
+  params: PriceQuoteParams
+  results: PromiseSettledResult<any>[]
+
+  constructor(message: string, params: PriceQuoteParams, results: PromiseSettledResult<any>[]) {
+    super(message)
+    this.params = params
+    this.results = results
+  }
+}
+
+type PriceSource = 'gnosis-protocol' | 'paraswap'
+type PriceInformationWithSource = PriceInformation & { source: PriceSource; data?: any }
+type PromiseRejectedResultWithSource = PromiseRejectedResult & { source: PriceSource }
+
+/**
+ *
+ * @returns The best price among GPv2 API or Paraswap one
+ */
+async function _getBestPriceQuote(params: PriceQuoteParams): Promise<PriceInformation> {
+  // Get price from all API: Gpv2 and Paraswap
+  const pricePromise = withTimeout(getPriceQuote(params), PRICE_API_TIMEOUT_MS, 'GPv2: Get Price API')
+  const paraSwapPricePromise = withTimeout(
+    getPriceQuoteParaswap(params),
+    PRICE_API_TIMEOUT_MS,
+    'Paraswap: Get Price API'
+  )
+
+  // Get results from API queries
+  const [priceResult, paraSwapPriceResult] = await Promise.allSettled([pricePromise, paraSwapPricePromise])
+
+  // Prepare an array with all successful estimations
+  const priceQuotes: Array<PriceInformationWithSource> = []
+  const errorsGetPrice: Array<PromiseRejectedResultWithSource> = []
+
+  if (isPromiseFulfilled(priceResult)) {
+    priceQuotes.push({ ...priceResult.value, source: 'gnosis-protocol' })
+  } else {
+    errorsGetPrice.push({ ...priceResult, source: 'gnosis-protocol' })
+  }
+
+  if (isPromiseFulfilled(paraSwapPriceResult)) {
+    const paraswapPrice = toPriceInformation(paraSwapPriceResult.value)
+    paraswapPrice && priceQuotes.push({ ...paraswapPrice, source: 'paraswap', data: paraSwapPriceResult.value })
+  } else {
+    errorsGetPrice.push({ ...paraSwapPriceResult, source: 'paraswap' })
+  }
+
+  if (errorsGetPrice.length > 0) {
+    const sourceNames = errorsGetPrice.map(e => e.source).join(', ')
+    console.error('[hooks::useRefetchPriceCallback] Some API failed or timed out: ' + sourceNames, errorsGetPrice)
+  }
+
+  if (priceQuotes.length > 0) {
+    const sourceNames = priceQuotes.map(p => p.source).join(', ')
+    console.log('[hooks::useRefetchPriceCallback] Get best price succeeded for ' + sourceNames, priceQuotes)
+    const amounts = priceQuotes.map(quote => quote.amount).filter(Boolean) as string[]
+
+    // Take the best price: Aggregate all the amounts into a single one.
+    //  - Use maximum of all the result for "Sell orders":
+    //        You want to get the maximum number of buy tokens
+    //  - Use minimum "Buy orders":
+    //        You want to spend the min number of sell tokens
+    const aggregationFunction = params.kind === OrderKind.SELL ? 'max' : 'min'
```

Comment:
nit: this isn't a function, i would name this `aggregationType` or `aggregationMethodName`